### PR TITLE
feat: [Validation] Callable Rules

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
For better compatibility to CI3 users. 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/system/Validation/ValidationTest.php [hunk 1]
```diff
@@ -343,6 +343,101 @@ public function testClosureRuleWithLabel(): void
...
+            ],
+        ]);
+
+        $data   = ['foo' => 'xyz'];
+        $result = $this->validation->run($data);
+
+        $this->assertFalse($result);
+        $this->assertSame(
+            ['foo' => 'The foo value is not "abc"'],
+            $this->validation->getErrors()
```
Comment (MGatner) [inline tests/system/Validation/ValidationTest.php:413]: I thought CS Fixer was enforcing newline parameter trailing commas?
@paulbalandan 
```suggestion
            $this->validation->getErrors(),
```

File: tests/system/Validation/ValidationTest.php [hunk 2]
```diff
@@ -343,6 +343,101 @@ public function testClosureRuleWithLabel(): void
...
+                ],
+            ],
+        ]);
+
+        $data   = ['secret' => 'xyz'];
+        $result = $this->validation->run($data);
+
+        $this->assertFalse($result);
+        $this->assertSame(
+            ['secret' => 'The シークレット is invalid'],
```
Comment (MGatner) [inline tests/system/Validation/ValidationTest.php:436]: やんちゃな

File: tests/system/Validation/ValidationTest.php (hunk 1 above)
Comment (kenjis) [reply tests/system/Validation/ValidationTest.php:413]: @MGatner It needs PHP 8.0 !
https://wiki.php.net/rfc/trailing_comma_in_parameter_list

Comment (paulbalandan) [reply tests/system/Validation/ValidationTest.php:413]: No. But there's a rule for that but disabled.

**Comment by kenjis** [comment]
Rebased to resolve conflicts in the docs.

File: tests/system/Validation/ValidationTest.php (hunk 1 above)
Comment (MGatner) [reply tests/system/Validation/ValidationTest.php:413]: Oh wow. I've been working in 8 so much I totally forgot this was new 😅 Thanks team.

**Review by MGatner — APPROVED** [review]

**Pull request merged**